washee_app/lib/features/booking/presentation/provider/calendar_provider.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/material.dart';
+import 'package:washee/core/helpers/date_helper.dart';
+
+class CalendarProvider extends ChangeNotifier {
+  bool _isLoadingDays = false;
+  final List<int> _numberOfFields = [
+    1,
+    2,
+    3,
+    4,
+    5,
+    6,
+    7,
+    8,
+    9,
+    10,
+    11,
+    12,
+    13,
+    14,
+    15,
+    16,
+    17,
+    18,
+    19,
+    20,
+    21,
+    22,
+    23,
+    24,
+    25,
+    26,
+    27,
+    28,
+    29,
+    30,
+    31
+  ];
+
+  final List<int> _greenScoreData = [
+    -1,
+    -1,
+    -1,
+    -1,
+    -1,
+    1,
+    -1,
+    10,
+    -1,
+    -1,
+    -1,
+    -1,
+    -1,
+    2,
+    1,
+    2,
+    3,
+    4,
+    5,
+    6,
+    7,
+    8,
+    9,
+    10,
+    -1,
+    -1,
+    -1,
+    -1,
+    -1,
+    -1,
+    -1,
+  ];
+
+  Map<String, List<DateTime>> daysInMonthMap = {};
+
+  // Fills up the daysInMonthMap with the respective number of dates for the given month
+  getDaysInBetween(DateTime startDate) {
+    var endDate = DateTime(startDate.year, startDate.month + 1, 0);
+
+    daysInMonthMap[_dateHelper.monthName(startDate.month)] = [];
+
+    if (daysInMonthMap[_dateHelper.monthName(startDate.month)] != null) {
+      for (int i = 0; i <= endDate.difference(startDate).inDays; i++) {
+        daysInMonthMap['${_dateHelper.monthName(startDate.month)}']!
+            .add(startDate.add(Duration(days: i)));
```

Comment:
her kunne jeg godt tænke mig at vi anvender getFieldNumberDate(), 
lav evt. en funktion til at beregne antal fields. således at dage længst til venstre ALTID er mandage, og længst til Højre altid er søndage.
noget i stil med
 fields = ceil(offset+monthLength/7)*7

hvis første dag er en Onsdag så er offset 2 fordi mandag og tirsdag skal tilføjes som tomme datoer

Reply:
Det er nice to have. Funktionalitet over UI lige nu. Specielt når det kun drejer sig om at række en container ind :-)